src/core/lib/event_engine/posix_engine/event_poller_posix_default.cc

```diff
@@ -55,12 +100,18 @@ PosixEventPoller* MakeDefaultPoller(Scheduler* scheduler) {
       poller = MakePollPoller(scheduler, /*use_phony_poll=*/true);
     }
   }
+  if (!std::exchange(g_registered, true)) {
+#ifdef GRPC_POSIX_FORK_ALLOW_PTHREAD_ATFORK
+    pthread_atfork(grpc_prefork, grpc_postfork_parent, grpc_postfork_child);
+#endif  // GRPC_POSIX_FORK_ALLOW_PTHREAD_ATFORK
+  }
+  g_forkable_instances->emplace_back(poller);
```

Comment:
this can also live inside the pthread guards.

Reply:
Done.